common/variant/src/main/java/org/apache/spark/types/variant/Variant.java

```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package org.apache.spark.variant;
+package org.apache.spark.types.variant;
 
 /**
  * This class is structurally equivalent to {@link org.apache.spark.unsafe.types.VariantVal}. We
```

Comment:
New package name follows the existing `org.apache.spark.unsafe.types.VariantVal`pattern by adding `types`.